nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/ListenTrapSNMP.java

```diff
@@ -99,29 +147,73 @@ public class ListenTrapSNMP extends AbstractSessionFactoryProcessor {
     )));
 
     private volatile SNMPTrapReceiverHandler snmpTrapReceiverHandler;
+    private volatile List<UsmUser> usmUsers;
+
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
```

Comment:
Setting the USM users requires an O(n) operation that may hit the disk.
We could argue that n (number of users) will never be exceedingly large and reading one small file from the disk should not take long.
Still, in general validation - depending on the flow - may occur 1000s of times every 5 seconds.

I would move this logic to the user-initiated verification and into the @OnScheduled method.